datafusion/common/src/config.rs

```diff
@@ -467,6 +472,7 @@ config_namespace! {
 
         /// (writing) Sets max statistics size for any column. If NULL, uses
         /// default parquet writer setting
+        #[allow(deprecated)]
```

Comment:
It hink the idea is to mark this as deprecated (not mark it as allowed deprecated)

Like
```suggestion
        #[deprecated(since = "45.0.0", note = "Setting does not do anything")]
```

Reply:
Opening a new PR #14188  since this commit tree is a bit dirty

Reply:
> Thanks for the review, Opening a new PR #14188 to fix this mess, since this commit tree is a bit dirty

FYI when we merge to `main` in this repository it is squashed to a single commit (aka having a dirty commit history on the PR is ok)